Sources/PackageRegistry/RegistryManager.swift

```diff
@@ -0,0 +1,253 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import PackageLoading
+import PackageModel
+
+import TSCBasic
+import TSCUtility
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+public enum RegistryError: Error {
+    case invalidOperation
+    case invalidResponse
+    case invalidURL
+    case invalidChecksum(expected: String, actual: String)
+    case invalidPackage
+}
+
+public final class RegistryManager {
+    public static var configuration: URLSessionConfiguration = {
+        var configuration = URLSessionConfiguration.default
+        configuration.httpAdditionalHeaders = [
+            "User-Agent": "Swift Package Manager",
+        ]
+
+        return configuration
+    }()
+
+    private lazy var session = URLSession(configuration: Self.configuration)
```

Comment:
what is our plan for linux? pulling in FoundationNetworking means adding a dependency on OpenSSL 😭

Reply:
this will also be a problem for the new package collections feature, so I feel we ay need to solve this by creating an "HTTPClient" abstraction which addresses this issue and some of the following comments.

Reply:
We have an existing semi-abstraction in https://github.com/apple/swift-tools-support-core/blob/main/Sources/TSCUtility/Downloader.swift -- we should consolidate all these.

Reply:
In general, I'm -1 on the idea of consolidating to a new networking abstraction (see https://github.com/apple/swift-package-manager/pull/3040#pullrequestreview-527477281). 

If `Downloader` is [currently importing `FoundationNetworking`](https://github.com/apple/swift-tools-support-core/blob/7fa19735d36bcf30244ab23ebca35b2bf479837a/Sources/TSCUtility/Downloader.swift#L14) doesn't that mean we already have a dependency on OpenSSL? My first implementation used `AsyncHTTPClient`, but I [reimplemented using `URLSession`](8bcb6b4f7e185dda3f394910f9ae9444712b72c6) to match what was already there.

Reply:
excellent points. more thoughts on https://github.com/apple/swift-package-manager/pull/3040#issuecomment-724951369